### DISCUSSION TITLE
how to specify python environment?

### DISCUSSION THREAD

**Discussion by shi-yan** [PyO3/pyo3#2206] [Questions]
I have a question regarding this document: https://pyo3.rs/v0.16.0/python_from_rust.html

I'm thinking about creating a CI/CD system. Users can create python scripts, and a rust-based management system runs those scripts. I hope to use pyo3 to implement it. But I'm not sure how pyo3 selects the python environment.

For examples, if I have multiple python environments installed on my host. Only one of them has the package pytorch installed, how can I be sure that pyo3 will find this python environment? Is this something we can specify programmatically?

**Comment by davidhewitt** [comment]
Thanks for the question. It's a bit of an open question that we need to explore.

I think you can use the `PYTHONHOME` and `PYTHONPATH` environment variables to control a lot of what you want. However, more research in this area is welcome. I know from firsthand experience that on Linux that running PyO3 inside a virtualenv "just works", however Windows and macOS are less clear. See #1896 #1741 #1554. If you're willing to do any investigation into this, I'd be happy to add that to our documentation.

**Reply by l-monninger** [reply]
Did anyone further investigate running on MacOs?

**Reply by davidhewitt** [reply]
Not as far as I am aware, if someone is willing to champion this with documentation or other work we need on the PyO3 side, I would be very grateful!

**Reply by l-monninger** [reply]
I might have some time next week.
